feat(Product) : Product detail 및 review 조회

Kim-SeonBeom opened the pull request:
## 변경사항
ProductCategory    -  @Getter 어노테이션 추가
Review                    -  userName 컬럼 추가

product_detail        -  controller,service 및 test코드 작성
review                     -  controller,service 및 test코드 작성
## 체크 리스트 (Checklist)

pull request 전에 아래 체크 리스트들을 만족하는 지 확인한 후 체크('x') 표시를 해주시기 바랍니다.

- [x]  master 브랜치가 아니라 **develop** 브랜치에 Merge하도록 pull request를 작성 중이신가요?
- [x]  모든 **테스트**가 성공했나요?

Commit:
ProductCategory    -  @Getter 어노테이션 추가
Review             -  userName 컬럼 추가

product_detail     -  controller,service 및 test코드 작성

review             -  controller,service 및 test코드 작성

Ko-udon commented on src/main/java/com/zerobase/wishmarket/domain/product/model/type/ProductCategory.java:
```diff
@@ -1,8 +1,10 @@
 
 @RequiredArgsConstructor
+@Getter
 public enum ProductCategory {
```
카테고리 (기타) 에 대해서 `"ETC(6)"` 를 정의했었는데 혹시 지워졌을까요?

Kim-SeonBeom replied:
아래 위아래 모양인 확장버튼 누르시면 확인하실수 있습니다

Ko-udon commented on src/main/java/com/zerobase/wishmarket/domain/product/service/ProductService.java:
```diff
@@ -1,14 +1,26 @@
+            throw new ProductNotFoundException();
+        }
+        Product product = optionalProduct.get();
+        return ProductDetailDto.of(product);
```
`return ProductDetailDto.of(optionalProduct.get())` 로 줄일수 있을것 같아요!

Kim-SeonBeom replied:
감사합니다 이 부분 수정하겠습니다.

Commit:
fix(product) : PR내용 반영
ProductService 수정

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/service/ProductService.java:
```diff
@@ -1,14 +1,26 @@
+    public ProductDetailDto detail(Long productId) {
+        Optional<Product> optionalProduct = productRepository.findById(productId);
+        if (!optionalProduct.isPresent()) {
+            throw new ProductNotFoundException();
```
Exception은 ErrorCode로만 관리되고 따로 클래스는 안만드는 쪽으로 구현해주세요

Kim-SeonBeom replied:
협의 내용을 잊어버렸네요  수정하겠습니다

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/service/ReviewService.java:
```diff
@@ -0,0 +1,20 @@
+    private final ReviewRepository reviewRepository;
+
+    public Page<Review> reviews(Long productId, Integer page, Integer size) {
+        PageRequest pageRequest = PageRequest.of(page-1, size);
```
왜 page -1을 하시나요??

Kim-SeonBeom replied:
보통 사용하는 웹 페이지에서 페이지는 1페이지 부터 시작하지만 
페이징처리를 했을때에는 페이지의 시작은 0페이지부터 시작하기때문에 -1을 해주었습니다.
이 부분은 FE담당자와 상의를 해보겠습니다.

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/service/ProductService.java:
```diff
@@ -1,14 +1,26 @@
-
-
+    public ProductDetailDto detail(Long productId) {
+        Optional<Product> optionalProduct = productRepository.findById(productId);
```
.orElseThrow() 로 error 반환하고 Entity를 바로 가져올 수 있지 않을까요???

Kim-SeonBeom replied:
해당부분 반영하여 수정하였습니다

Commit:
fix(product) : product_service 수정 그에 따른 test 코드 변경

1. Optional<Product> -> Product 객체로 직접반환
2. Exception 발생 시 코드 수정
3. 변경 사항에 따른 test 코드 수정
4. OAuthLoginTest 주석처리 해제

Commit:
fix(product) : reviewService 수정 및 변경사항에 따른 test수정

1.review 페이징처리에 대한 page size 변경(12개)
2.reviewService 수정에 따른 test코드 수정

Commit:
Merge branch 'dev' into #33PRODUCT-DETAIL
